packages/jsii-reflect/lib/tree.ts

```diff
@@ -0,0 +1,278 @@
+import colors = require('colors/safe');
+import { AsciiTree } from 'oo-ascii-tree';
+import { Assembly } from './assembly';
+import { ClassType } from './class';
+import { Dependency } from './dependency';
+import { EnumType } from './enum';
+import { InterfaceType } from './interface';
+import { Method } from './method';
+import { Parameter } from './parameter';
+import { Property } from './property';
+import { TypeReference } from './type-ref';
+import { TypeSystem } from './type-system';
+
+export interface TypeSystemTreeOptions {
+  /**
+   * Show all entity types (supersedes other options)
+   * @default false
+   */
+  showAll?: boolean;
+
+  /**
+   * Show type members (methods, properties)
+   * @default false
+   */
+  members?: boolean;
+
+  /**
+   * Show dependencies
+   * @default false
+   */
+  dependencies?: boolean;
+
+  /**
+   * Show inheritance information (base classes, interfaces)
+   * @default false
+   */
+  inheritance?: boolean;
+
+  /**
+   * Show types
+   * @default false
+   */
+  types?: boolean;
+
+  /**
+   * Show method signatures.
+   * @default false
+   */
+  signatures?: boolean;
+
+  /**
+   * Output with ANSI colors
+   * @default true
+   */
+  colors?: boolean;
+}
+
+/**
+ * Visualizes a `TypeSystem` as an ASCII tree.
+ */
+export class TypeSystemTree extends AsciiTree {
+  constructor(typesys: TypeSystem, options: TypeSystemTreeOptions = { }) {
+    super();
+
+    if (options.showAll) {
+      options.dependencies = true;
+      options.inheritance = true;
+      options.members = true;
+      options.signatures = true;
+      options.types = true;
+    }
+
+    const shouldUseColors = options.colors === undefined ? true : options.colors;
+    const colorsEnabled = colors.enabled;
+    try {
```

Comment:
Would wrap the try-finally in a helper function (but I guess it's also good as-is). Reason I'm saying is because I was initially confused by `shouldUseColors` and `colorsEnabled`, and had to read down almost 20 lines to understand why `colorsEnabled` exists.

Reply:
done